### DISCUSSION TITLE
How to know that binding local address is done?

### DISCUSSION THREAD

**Discussion by simon-fu** [libp2p/rust-libp2p#4450] [General]
If app listens only on local, how to know that all of SwarmEvent::NewListenAddr have received?

**Comment by kehiy** [comment]
when the binding process is complete, you need to use the `SwarmEvent::NewListenAddr` event, which is emitted by the `Swarm::poll` method. This event is emitted when a new local listen address has been successfully bound.

**Reply by simon-fu** [reply]
Yes, I do know what SwarmEvent::NewListenAddr event means. My question is how do I know that all of SwarmEvent::NewListenAddr events have been emitted ? Is there any event like "SwarmEvent::LocalBindFinished" ?

**Reply by kehiy** [reply]
no, that seems there is no `SwarmEvent::LocalBindFinished`!

**Reply by thomaseizinger** [reply]
What does "done" mean? Most operating systems support dynamic creation and removal of interfaces so how would we know when we are "done"?

We will dynamically bind to all available interfaces as they become available if you listen on a wildcard address. If you don't want that, you should bind to a specific interface.

**Reply by simon-fu** [reply]
Got it. Thanks a lot.
